test/0.8.9/utils/pausable-util.test.sol

```diff
@@ -0,0 +1,186 @@
+        assertEq(pausableUntil.isPaused(), false);
+
+        // pause until _resumeSince
+        pausableUntil.setPausedState(_resumeSince);
+        assertEq(pausableUntil.isPaused(), true);
+
+        // check a random point in time (either before or after _resumeSince)
+        vm.warp(_random_time);
+        if (_random_time < _resumeSince) {
+            assertEq(pausableUntil.isPaused(), true);
+        } else {
+            assertEq(pausableUntil.isPaused(), false);  
+        }
+
+        // go forward in time to _resumeSince - 1 block, check edge
+        vm.warp(_resumeSince - 1);
+        assertEq(pausableUntil.isPaused(), true);
+
+        // go forward in time to _resumeSince block, check edge
+        vm.warp(_resumeSince);
+        assertEq(pausableUntil.isPaused(), false);
+    }
+
+    function testPauseUntil(uint256 _pauseUntilInclusive, uint256 _random_time) public {
+        vm.assume(_pauseUntilInclusive != 0);
+        vm.assume(_random_time != 0);
+        vm.assume(_random_time != _pauseUntilInclusive);
+                
+        if (_pauseUntilInclusive < block.timestamp) {
+            vm.expectRevert();
+            pausableUntil.pauseUntil(_pauseUntilInclusive);
+            assertEq(pausableUntil.isPaused(), false);
+        } else {
+            pausableUntil.pauseUntil(_pauseUntilInclusive);
+            assertEq(pausableUntil.isPaused(), true);
+        }
+
+        // check a random point in time (either before or after _pauseUntilInclusive)
+        vm.warp(_random_time);
+        if (_random_time < _pauseUntilInclusive) {
+            assertEq(pausableUntil.isPaused(), true);
+        } else {
+            assertEq(pausableUntil.isPaused(), false);  
+        }
+    }
+
+    function testPauseFor(uint256 _duration, uint256 _random_time) public {
+        vm.assume(_random_time != 0);
+        assertEq(pausableUntil.isPaused(), false);
+
+        uint256 origin_block_time = block.timestamp;
+
+        if (_duration == 0) {
+            vm.expectRevert();
+            pausableUntil.pauseFor(_duration);
+        } else {
+            pausableUntil.pauseFor(_duration);
+            assertEq(pausableUntil.isPaused(), true);
+        }
+
+        // check a random point in time (either before or after _pauseUntilInclusive)
+        vm.warp(_random_time);
+
+        // bypass overflow in our own test arithmetic
+        unchecked {
+            uint256 sum = origin_block_time + _duration;
+            vm.assume(sum >= origin_block_time);
+            vm.assume(sum >= _duration);
+        }
+
+        if (_random_time < (origin_block_time + _duration)) {
+            assertEq(pausableUntil.isPaused(), true);
+        } else {
+            assertEq(pausableUntil.isPaused(), false);  
+        }
+    }
+
+    function testResumeWhenPaused() public {
+        assertEq(pausableUntil.isPaused(), false);
+
+        // pause for the next 3 seconds
+        pausableUntil.setPausedState(block.timestamp + 3);
```

Comment:
would be great to not use a hardcoded constant because the real network has a twelve seconds cadence
(can fuzz it, maybe?)